Samples/NetworkMediaStreamer/CameraRTPStreamerApp/CameraRTPStreamerApp.cpp

```diff
@@ -0,0 +1,430 @@
+#include <windows.media.h>
+#include <windows.media.core.interop.h>
+#include <windows.foundation.h>
+#include <windows.Storage.streams.h>
+#include <winrt\base.h>
+#include <winrt\Windows.Media.Capture.h>
+#include <winrt\Windows.Media.Capture.Frames.h>
+#include <winrt\Windows.Graphics.Imaging.h>
+#include <winrt\Windows.Foundation.h>
+#include <winrt\Windows.Devices.Enumeration.h>
+#include <winrt\Windows.Media.Devices.h>
+#include <winrt\Windows.Media.MediaProperties.h>
+#include <winrt\Windows.Networking.Connectivity.h>
+#include <winrt\Windows.storage.streams.h>
+
+#include "..\Common\inc\NetworkMediaStreamer.h"
+#include "..\Common\inc\RTPMediaStreamer.h"
+#include "..\Common\inc\RTSPServerControl.h"
+using namespace winrt;
+using namespace Windows;
+using namespace Media::Capture;
+using namespace Frames;
+using namespace Windows::Graphics::Imaging;
+using namespace Windows::Foundation;
+using namespace Windows::Devices::Enumeration;
+using namespace Windows::Media;
+using namespace Windows::Media::Devices;
+using namespace Windows::Media::MediaProperties;
+constexpr uint16_t ServerPort = 8554;
+constexpr uint16_t SecureServerPort = 6554;
+
+// Uncomment the following if you want to use FrameReader API instead of the Record-to-sink APIs
+//#define USE_FR 
+
+// sample test code to get localhost test certificate
+std::vector<PCCERT_CONTEXT> getServerCertificate()
+{
+    std::vector<PCCERT_CONTEXT> aCertContext;
+    PCCERT_CONTEXT pCertContext = NULL;
+    //-------------------------------------------------------
+    // Open the My store, also called the personal store.
+    // This call to CertOpenStore opens the Local_Machine My 
+    // store as opposed to the Current_User's My store.
+
+    auto hMyCertStore = CertOpenStore(CERT_STORE_PROV_SYSTEM,
+        X509_ASN_ENCODING,
+        0,
+        CERT_SYSTEM_STORE_LOCAL_MACHINE,
+        L"MY");
+
+    if (hMyCertStore == NULL)
+    {
+        printf("Error opening MY store for server.\n");
+        goto cleanup;
+    }
+    //-------------------------------------------------------
+    // Search for a certificate with some specified
+    // string in it. This example attempts to find
+    // a certificate with the string "example server" in
+    // its subject string. Substitute an appropriate string
+    // to find a certificate for a specific user.
+    do
+    {
+        pCertContext = CertFindCertificateInStore(hMyCertStore,
+            X509_ASN_ENCODING,
+            0,
+            CERT_FIND_SUBJECT_STR_W,
+            L"localhost", // use appropriate subject name
+            pCertContext
+        );
+        if (pCertContext)
+        {
+            aCertContext.push_back(CertDuplicateCertificateContext(pCertContext));
+        }
+    } while (pCertContext);
+    if (aCertContext.empty())
+    {
+        printf("Error retrieving server certificate.");
+    }
+cleanup:
+    if (hMyCertStore)
+    {
+        CertCloseStore(hMyCertStore, 0);
+    }
+    return aCertContext;

```

Comment:
can this be done with scope_exit we don't need use goto Cleanup?

Reply:
i think scope_exit is still in std::experimental for c++17
i cannot use wil due to nuget pack hurdles in cmake as mentioned earlier. 
However, i shall get rid of the goto cleanup , as it was not necessary.

---
In reply to: [494092998](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r494092998) [](ancestors = 494092998)